src/app/components/table/table.component.ts

```diff
@@ -1,20 +1,26 @@
-import { Component, OnInit, ChangeDetectionStrategy, Input, Output, EventEmitter } from '@angular/core';
+import { Component, OnInit, Input, Output, EventEmitter, OnDestroy } from '@angular/core';
 import { Element, ElementType, MainService } from 'src/app/services/main.service';
+import { Subject } from 'rxjs';
+import { map, takeUntil } from 'rxjs/operators';
 
 @Component({
-  selector: 'ui-table',
+  selector: 'app-ui-table',
   templateUrl: './table.component.html',
   styleUrls: ['./table.component.scss']
 })
-export class TableComponent implements OnInit {
+export class TableComponent implements OnInit, OnDestroy {
+  private readonly toDestroy$ = new Subject<void>();
+
   @Input()
   public set elements(elements: Array<Element>) {
+    if (!elements) { return; }
```

Comment:
Needed to check whether the incoming prop has values here so we can perform out task more safely, if there are no values, the function calls cannot be run as `undefined.forEach` or something like that.